crates/core/plugin_sm/src/plugin.rs

```diff
@@ -105,14 +106,22 @@ pub trait Plugin {
             };
             let module_url = module.url.clone();
             if let Some(url) = module_url {
-                match Self::download_from_url(module, &url, logger, download_path, self.identity())
+                if !url.url.is_empty() {
```

Comment:
As the same logic is repeated here, I would add a method `DownloadInfo::is_empty(&self)`.

Or better to solve the issue when the modules are deserialized or built, replacing any `Some(empty_url)` by `None`.